### PR TITLE
Request: Track fulfillment status

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
On web and in history:
image

### CODE REVIEW

**Review by dcposch — COMMENTED** [review]
Looks fantastic. Will review in more detail soon

File: apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx [hunk 1]
```diff
@@ -72,6 +73,13 @@ function TransferBody({ op }: { op: TransferOpEvent }) {
   const [account] = useAccount();
   assert(account != null);
 
+  const opRequestId = op.nonceMetadata
+    ? DaimoNonceMetadata.fromHex(op.nonceMetadata).identifier.toString()
+    : undefined;
+  const matchingTrackedRequest = account.trackedRequests.find(
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx:79]: 👍

File: apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx [hunk 2]
```diff
@@ -72,6 +73,13 @@ function TransferBody({ op }: { op: TransferOpEvent }) {
...
+    (req) => req.requestId === opRequestId && req.amount === `${op.amount}`
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx:80]: might be worth also requiring that the recipient is us

File: apps/daimo-mobile/src/view/screen/link/NoteScreen.tsx [hunk 3]
```diff
@@ -84,7 +85,11 @@ function NoteDisplay({
   );
 
   const [senderName, setSenderName] = useState("");
-  const nonce = new DaimoNonce(new DaimoNonceMetadata());
+  const nonceMetadata = new DaimoNonceMetadata(DaimoNonceType.ClaimNote);
```
Comment (dcposch) [inline apps/daimo-mobile/src/view/screen/link/NoteScreen.tsx:88]: 👍

File: packages/daimo-userop/src/nonce.ts [hunk 4]
```diff
@@ -29,20 +29,46 @@ export class DaimoNonce {
...
 
+export enum DaimoNonceType {
+  Send = 0,
+  CreateNote = 1,
+  ClaimNote = 2,
+  RequestResponse = 3,
+  MAX = 255, // At most one byte
+}
+
+export const MAX_NONCE_ID_SIZE_BITS = 56n;
```
Comment (dcposch) [inline packages/daimo-userop/src/nonce.ts:48]: fits into a int64 👍

**Review by dcposch — APPROVED** [review]
LGTM

recommend adding unit tests for DaimoNonce (generate, serialize, deserialize, ...) either in this or a follow up PR

**Comment by nalinbhardwaj** [comment]
> recommend adding unit tests for DaimoNonce (generate, serialize, deserialize, ...) either in this or a follow up PR

done!

File: apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx (hunk 2 above)
Comment (nalinbhardwaj) [reply apps/daimo-mobile/src/view/screen/HistoryOpScreen.tsx]: done

**Pull request merged**